### ISSUE TITLE
End-to-End Time-Lapse Video Synthesis from a Single Outdoor Image

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#26]
![image](https://user-images.githubusercontent.com/46308407/104124728-a5185580-5395-11eb-9770-bd1464ed517a.png)

## Key Idea

- Generate Time Lapse video NOT through some temporal model, but simply through **conditional  GAN** using actual time stamp as control variable.
- Use two datasets, AMOS with time stamp labels but poor in quality and TLVDB without time stamp but good quality, in ***semi-supervised fashion*** to generate interesting result. ***(Multi-Domain Training)***
- Create images of low resolution and upsample with ***automatic guided upsampling*** technique.

## Previous Approaches

- **Image & Video Stylization**

(ex) Neural Style Transfer, CycleGAN, etc.

⇒ Change the illumination in ***continuous manner!!***

- **Animating Still Images**

Future frame prediction task (GANs, VAEs)

Very short or fixed length sequence (Capability Issue)

⇒ Animate High resolution outdoor image of any length

- **Learning from Video Dataset**

Data size is large and less structured compared to still images

## Formulation

(problem) Generate $\hat{I}_t = G(I, t)$ (conditioned on single input image and timestamp)

$V = {\hat{I_0}, \hat{I_1}, ... , \hat{I_n}}$ is the output video

### Dataset

*AMOS*: Images taken with surveillance cameras with time stamps. Labeled, but not interesting.

*TLVDB*: Time Lapse videos of mostly landmarks with hundreds of frames taken with good cameras.  (unlabeled)

### Overall Architecture

![image](https://user-images.githubusercontent.com/46308407/104124738-b19cae00-5395-11eb-8210-c3601cd8122e.png)

MFJCG: Multi-Frame Joint Conditional Generation Network

(—————————Multi-Frame Joint Conditional Generation——————-————)

- First train $D_A$ with AMOS dataset. (For S*emi-supervised Learning*)
    - $G_A$, $D_A$ are all Multi-Frame Joint Conditional Generation networks.
    - Details ensue. **** (Multi-Frame Joint Conditional Generation)

(—————————————-—Multi-Domain Training——————————-————)

- $G_B$ generate samples from TLVDB with randomly initialized timestamps.
- Outputs of $G_B$ are domain-adapted with $G_A$, a U-NET responsible for transformation.
- Sequence of images are now in the same domain as previously trained → $D_A$ can work!
    - Evaluates both fidelity of individual images and quality of conditioned sequence.
- Meanwhile, $D_B$ only measures the fidelity of samples generated from $G_B$, before domain adaptation.

### Multi-Frame Joint Conditional Generation

![image](https://user-images.githubusercontent.com/46308407/104124744-bc574300-5395-11eb-8ea9-754fda18d003.png)

- Pretrained VGG-16 was used for Generator encoder.
- Discriminator evaluates two losses: ***unconditional(image level)*** and ***conditional(sequence level)***

![image](https://user-images.githubusercontent.com/46308407/104124749-c4af7e00-5395-11eb-9a70-8a2aed9f4473.png)

- Second term in $l_c$ forces discriminator to output $0$ for permuted (unmatched) sample

## Multi-Domain Training

- Loss Function

![image](https://user-images.githubusercontent.com/46308407/104124755-ced17c80-5395-11eb-9b0c-3dbbb0113db6.png)

## Guided Upsampling

- Produce lower resolution samples first, and use automatic guided upsampling.

Map Input image to Final output by per-pixel linear transformation with raw network output as guidance.

$\bar{I}(p) = T(p)(I(p)) = s(p) \times I(p) + b(p)$, $p$는 pixel, $\bar{I}$: Final Output, $\hat{I}$: raw output, $I$: input

Find $s(p), b(p)$ with ***LSE***

![image](https://user-images.githubusercontent.com/46308407/104124759-d7c24e00-5395-11eb-901d-0bb50bb6f8e3.png)

$N(p)$ is one-ring neighborhood of $p$

$w(I(p), I(q))$ is inverse color distance between neighboring pixels, $p,q$ of original input image

## Experiment Results

- Quantitative Comparison

User Study (Real or Fake?)

- Qualitative Comparison

![image](https://user-images.githubusercontent.com/46308407/104124767-e3157980-5395-11eb-861d-30fc9ee21b01.png)

![image](https://user-images.githubusercontent.com/46308407/104124774-ead51e00-5395-11eb-9f3b-6ded844e709d.png)

## Comments

- Interesting how they approached video generation task via conditional GAN model, simply using time stamp as condition input.
- Wonder how they encoded the timestamp as feature map.
- Smart how they concatenated Gaussian Noise to Generator bottleneck to separate between sequences.
- The way they used semi-supervised learning to fully utilize the best of given dataset was impressive.
- The loss function looks quite complex, and wonders if the training process is stable enough.
- Overall, I think the displayed samples are somewhat cherry-picked, as there aren't sufficient samples with good quality shown here.
- Apart from the content, the style of writing is not very refined. Quite difficult to read, especially figures. Would've been better if they included tables for quantitative results too.
- Think the guided upsampling is a little unnatural. Why didn't they just use progressive growing? Is that enough? (I will try coding one for myself)
